### PR TITLE
Added iterator to for loop

### PR DESCRIPTION
**Pull request by matthiasdebernardini**
this change removes a function and hoists some of the loop logic into the iterator

### CODE REVIEW

File: src/nsyt/remote.rs [hunk 1]
```diff
@@ -56,19 +44,16 @@ async fn process_results(
...
-        valid_img_count += 1; //ITM is an anti-pattern. TODO: enumerate(), and have the 'e' fall out.
+    for (idx, image) in results
+        .items
+        .iter()
+        .filter(|image| {
+            image.is_thumbnail && !thumbnails && search.iter().any(|i| image.imageid.contains(i))
+        })
+        .enumerate()
+    {
+        valid_img_count = idx as i32; //ITM is an anti-pattern. TODO: enumerate(), and have the 'e' fall out.
```
Comment (matthiasdebernardini) [inline src/nsyt/remote.rs:55]: what you want for the valid img count var is the length of the iterator

Comment (kmgill) [reply src/nsyt/remote.rs:55]: Any changes made here can _probably_ be made to `src/m20/remote.rs` and `src/msl/remote.rs`

This does look better than previous.

File: src/enums.rs [hunk 2]
```diff
@@ -47,6 +47,7 @@ pub enum Instrument {
     M20HeliRte,
     NsytICC,
     NsytIDC,
+    #[default]
```
Comment (kmgill) [inline src/enums.rs:50]: This is good, I didn't know about `Default` for enums. (perhaps that's what you're doing below with the empty initialization?)   I might get rid of `None` as an enum value and replace with `Option<Instrument>`

File: src/m20/zcam.rs [hunk 3]
```diff
@@ -70,24 +70,27 @@ impl Calibration for M20MastcamZ {
         }
 
         let mut warn = false;
-        let mut instrument = Instrument::M20MastcamZLeft;
+        let instrument;
```
[the remark this answers is not in the document]
Comment (matthiasdebernardini) [reply src/m20/zcam.rs:73]: Yeah this change is purely about readability.

Default implementations tend to be more of an opinion/preference thing regarding how types are implemented. They can come in handy so it's nice to have them around. 

Could also make the variable mutable, initialize it to Deftault and then reassign it later on. That is wordier, so I prefer how it stays. Having default can come in handy sometimes, is easy enough to add and if nobody uses it then it gets optimized away. 

If you end up making a common-core type library ever,  then its nice to have.

File: src/msl/ecam.rs [hunk 4]
```diff
@@ -42,38 +42,69 @@ impl Calibration for MslEcam {
...
-                // Left
-                instrument = enums::Instrument::MslNavCamLeft;
-            } else {
-                // Assume Right
-                instrument = enums::Instrument::MslNavCamRight;
+        let instrument;
+        match util::filename_char_at_pos(input_file, 0) {
+            'N' => {
+                // NAVCAMS
+                match util::filename_char_at_pos(input_file, 1) == 'L' {
```
[the remark this answers is not in the document]
Comment (matthiasdebernardini) [reply src/msl/ecam.rs:49]: I'm not either, but then again I was not wild about the embedded if statements.

File: src/enums.rs (hunk 2 above)
Comment (matthiasdebernardini) [reply src/enums.rs:50]: I just added it because it was simple enough and is nice to have.

File: src/nsyt/metadata.rs [hunk 5]
```diff
@@ -159,11 +170,11 @@ impl ImageMetadata for Image {
     }
 
     fn get_mast_az(&self) -> Option<f64> {
-        None
+        unimplemented!()
```
Comment (kmgill) [inline src/nsyt/metadata.rs:173]: Already merged, but am putting these back to `None`. `unimplemented!()` will cause a panic when the metadata is written back out to disk, which is done in a mission-agnostic manner and uses all fields. `mast_az` and `mast_el` don't apply to InSight and will always be none.

**Pull request merged**